### PR TITLE
[db] Add index to d_b_workspace_instance on startedTime & stoppingTime

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Adds missing indices to avoid full table scan.

[Internal Context](https://gitpod.slack.com/archives/C02EN94AEPL/p1662471052279859?thread_ts=1662468125.845989&cid=C02EN94AEPL)

## Related Issue(s)

Relates to #13060

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

/hold to run this manually as it is a long-running migration.

### CODE REVIEW

**Review by svenefftinge — APPROVED** [review]

File: components/gitpod-db/src/typeorm/migration/1662491259313-WorkspaceInstanceStartedStoppingIndex.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { indexExists } from "./helper/helper";
+
+export class WorkspaceInstanceStartedStoppingIndex1662491259313 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const TABLE_NAME = "d_b_workspace_instance";
+
+        const startedTimeIndex = "IDX_workspace_instance__started_time";
+        if (!(await indexExists(queryRunner, TABLE_NAME, startedTimeIndex))) {
+            await queryRunner.query(`CREATE INDEX ${startedTimeIndex} ON ${TABLE_NAME} (startedTime)`);
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/migration/1662491259313-WorkspaceInstanceStartedStoppingIndex.ts:16]: Do we need the magical `, ALGORITHM=INPLACE, LOCK=NONE` statement here?

@geropl

Comment (geropl) [reply components/gitpod-db/src/typeorm/migration/1662491259313-WorkspaceInstanceStartedStoppingIndex.ts:16]: > magical

Link to MySQL docs: https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl-operations.html

tl;dr: no, adding 2ndary indexes is a in-place operation that does not lock.

This is more relevant for adding primary keys, or re-naming (cmp. docs).

**Comment by easyCZ** [comment]
Removing auto-link to close https://github.com/gitpod-io/gitpod/issues/13060, just landing the PR doesn't actually fix the issue so we need to track manually.

**Comment by geropl** [comment]
@easyCZ We could merge this one now, right? :thinking:

**Comment by easyCZ** [comment]
@geropl Yep! I'll just create a new migration and move the contents of this one to ensure the TS is most recent. I'm unsure exactly how the migrations handle cases where the migrations in the past is inserted.

**Comment by geropl** [comment]
> I'm unsure exactly how the migrations handle cases where the migrations in the past is inserted.

It handles it well, e.g. does identity by name, and performs all which have not yet been executed. :+1:

**Comment by easyCZ** [comment]
@geropl Thanks. I've moved the file anyway just to keep it clean. 

/unhold
